### PR TITLE
Host install.txt on Github

### PR DESCRIPTION
**Pull request by jvonau**
Initial import for install.txt to be hosted at github
rework ordering for better future use of iiab-factory

### CODE REVIEW

**Comment by tim-moody** [comment]
is a monitor and keyboard required for this script

>  RUN "touch /boot/ssh" to enable incoming SSH after next reboot

is there any provision for exiting from an infinite loop of fail, reboot, rerun, like a maximum number of iterations?

**Comment by jvonau** [comment]
A monitor not required if you pre-prep the image with '/boot/ssh' then you could kick off the curl install over ssh and after the reboot the install continues to run without ssh. The installer runs once upon boot and will continue to retry upon each boot until all the installs are run then disables itself when complete.

**Comment by jvonau** [comment]
The script for the master branch works by using:
curl https://raw.githubusercontent.com/jvonau/iiab-factory/master/scripts/install-latest.txt | sudo bash
The install.txt file does not use the reboot and iiab-installer method it's intended to replace the current script that does not live under git control but needs the git branches named 'release-6.6' for iiab and iiab-admin-console to be created, or at least for iiab and the variable changed for iiab-admin console to be 'master'

**Comment by jvonau** [comment]
After the first reboot and iiab-installer is running the progress can be monitored with 'journalctl --no-pager -u iiab-installer' Once the install completes and reboots journalctl --no-pager -u iiab-installer' will show:
Sep 04 05:13:09 box.lan systemd[1]: Starting IIAB installer...
Sep 04 05:13:11 box.lan continue.sh[1249]: Removed /etc/systemd/system/multi-user.target.wants/iiab-installer.service.
Sep 04 05:13:11 box.lan systemd[1]: Started IIAB installer.

Now the service is disabled for subsequent reboots.

File: scripts/install-latest.txt [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+
+# replace with ansible role WIP
+cp $BASE/iiab-factory/roles/iiab-installer/files/iiab-installer.service /etc/systemd/system/
+systemctl enable iiab-installer
+
+cd $BASE/iiab/
+./iiab-install $REINSTALL
+apt update
+apt -y dist-upgrade
+reboot
```
Comment (holta) [inline scripts/install-latest.txt]: Doesn't `apt update; apt -y dist-upgrade; reboot` need to happen *prior to* `./iiab-install`, so the network stage[*] works reliably?

[*] equivalent to ./iiab-network (and potentially also other aspects of IIAB's install routine!)

The concern is that *many* implementers run "apt update; apt -y dist-upgrade" on a new OS (on their own, out of habit, whether we ask them to or not!) and yet do not reboot after that &mdash; leaving them with a partially installed kernel, that is not yet running &mdash; and then blocks their IIAB installation :-(

**Comment by jvonau** [comment]
The network stage is reliable when the there is no kernel mismatch between what has been installed and what is running. Using the running kernel will ensure there is no mismatch for the network stage, Delaying the reboot till after when a kernel might be installed makes life easier support wise but that new kernel install process could occur at any point in the install. Having the reboot take place at this point is to catch those who have installed a new kernel but have not rebooted, as stage 9 is complete, network is configured but if there was a failure at 'systemd restart networking' the services were not restarted but otherwise ready and working correctly with a reboot.

**Comment by holta** [comment]
PR https://github.com/iiab/iiab-factory/pull/55 implements this in a slightly different way.

**Pull request closed without merging**